airflow/contrib/hooks/spark_submit_hook.py

```diff
@@ -185,6 +185,8 @@ def _resolve_connection(self):
                 conn_data['master'] = "{}:{}".format(conn.host, conn.port)
             else:
                 conn_data['master'] = conn.host
+            if conn.uri:
+                conn_data['master'] = conn.uri
```

Comment:
since the specified URI might consist of other attributes other than scheme, host, and port (ex: query & schema), I think we couldn't directly assign the master address to `conn.uri`

Reply:
from my view, conn_data['master'] is equal to the MASTER_URL in `spark-submit` , the options are:
```
spark://host:port, mesos://host:port, yarn, k8s://https://host:port, or local (Default: local[*]).
```
`conn.uri`  probably comes from environment var `AIRFLOW_CONN_SPARK_DEFAULT`, which should be the same as one of options above, I guess.

Reply:
hmm but what if the `extra` attribute is needed? I mean, based on [this](https://github.com/apache/airflow/blob/master/airflow/contrib/hooks/spark_submit_hook.py#L190), we can retrieve additional `spark-submit` params, such as deploy-mode, spark-binary, etc.).

If the URI is exactly the same as the MASTER_URL, how to extract this `extra` values?

Reply:
Oh, thanks for your remind, I updated the code, taking the query params into consideration. Please review again.